src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,124 @@
+import com.databricks.labs.overwatch.env.{Database, Workspace}
+import org.apache.spark.sql.{Column, DataFrame, Dataset}
+import org.apache.log4j.{Level, Logger}
+
+
+object Snapshot extends SparkSessionWrapper {
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  import spark.implicits._
+
+  /**
+   * Create a backup of the Overwatch datasets
+   *
+   * @param snapshotRootPath prefix of path target to send the snap
+   * @param cloneLevel       Deep or Shallow
+   * @param asOfTS           appends asOfTimestamp option to Delta reader to limit data on clone. This will only go back as
+   *                         far as the latest vacuum by design.
+   * @param excludes         Array of table names to exclude from the snapshot
+   *                         this is the table name only - without the database prefix
+   * @return
+   */
+  private[overwatch] def snap(
+              bronze: Bronze,
+              workspace: Workspace,
+              snapshotRootPath: String,
+              cloneLevel: String = "DEEP",
+              asOfTS: Option[String] = None,
+              excludes: Array[String] = Array()
+            ): Unit = {
+      val acceptableCloneLevels = Array("DEEP", "SHALLOW")
+      require(acceptableCloneLevels.contains(cloneLevel.toUpperCase), s"SNAP CLONE ERROR: cloneLevel provided is " +
+        s"$cloneLevel. CloneLevels supported are ${acceptableCloneLevels.mkString(",")}.")
+
+      val bronzeTargets = bronze.targetToSnap
+      val cleanExcludes = excludes.map(_.toLowerCase).map(exclude => {
+        if (exclude.contains(".")) exclude.split("\\.").takeRight(1).head else exclude
+      })
+      val targetsToSnap = bronzeTargets
+        .filter(_.exists()) // source path must exist
+        .filterNot(t => cleanExcludes.contains(t.name.toLowerCase))
+
+      val cloneSpecs  = buildCloneSpecs(targetsToSnap,snapshotRootPath,cloneLevel,asOfTS)
+      val cloneReport = Helpers.parClone(cloneSpecs)
+      val cloneReportPath = if (snapshotRootPath.takeRight(1) == "/") s"${snapshotRootPath}report/" else s"${snapshotRootPath}/clone_report/"
+      cloneReport.toDS.write.format("delta").save(cloneReportPath)
+    }
+
+  /**
+   * Create a backup of the Overwatch datasets
+   *
+   * @param snapshotRootPath prefix of path target to send the snap
+   * @param excludes         Array of table names to exclude from the snapshot
+   *                         this is the table name only - without the database prefix
+   * @return
+   */
+
+  private[overwatch] def incrementalSnap(
+                       bronze: Bronze,
+                       workspace: Workspace,
+                       snapshotRootPath: String,
+                       excludes: Array[String] = Array()
+                     ): Unit = {
+
+    val bronzeTargets = bronze.targetToSnap
+    val cleanExcludes = excludes.map(_.toLowerCase).map(exclude => {
+      if (exclude.contains(".")) exclude.split("\\.").takeRight(1).head else exclude
```

Comment:
use the new path slash helpers introduced in #825 -- these are merged to 0720 now so rebase and you should have access to them.